### PR TITLE
Update node and dependencies

### PR DESCRIPTION
**Pull request by hatched**
After a number of previous branches fixed larger issues with the node upgrade we can finally upgrade to the latest node and the latest dependencies. This branch switches our reliance on the nodejs installation in the yellow ppa and instead pulls node directly from the nodesource deb.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -213,7 +213,7 @@ $(BUILT_D3):
 	  node_modules/d3/src/svg/line.js \
 	  node_modules/d3/src/svg/arc.js \
 	  node_modules/d3/src/event/drag.js \
-	  $(GUISRC)/app/assets/javascripts/unscaled-pack.js \
+		node_modules/d3/src/layout/pack.js \
```
Comment (hatched) [inline Makefile]: We will need to qa this once we have a charm installable GUI again to make sure that this functionality is either a) good enough or b) we have to write a new unscaled-pack. The methods our unscaled-pack relies on are no longer available in the latest version of d3

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3593/
Test FAILed.

File: jujugui/static/gui/src/test/test_app.js [hunk 2]
```diff
@@ -82,13 +82,12 @@ function injectData(app, data) {
 
     });
 
-    afterEach(function(done) {
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_app.js]: The updated mocha doesn't like it when done() is called multiple times per test

File: package.json [hunk 3]
```diff
@@ -10,34 +10,33 @@
   },
   "dependencies": {
     "babel": "^5.8.29",
-    "chai": "^3.2.0",
```
Comment (hatched) [inline package.json]: Updates almost all of the dependencies. babel still needs to be updated to the 6.x line but the changes are quite big so I will do that on another friday

**Comment by bac** [comment]
Please make sure the makefile help includes all the targets. I noticed fast-babel is missing.

**Comment by bac** [comment]
So QA was not a clear win. @hatched please advise about the resource issues.

**Pull request closed without merging**